presto-password-authenticators/src/test/java/io/prestosql/plugin/password/TestFileAuthConfig.java

```diff
@@ -0,0 +1,274 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.password;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import io.airlift.log.Logger;
+import io.airlift.units.Duration;
+import io.prestosql.plugin.password.file.EncryptionUtil;
+import io.prestosql.plugin.password.file.FileBasedAuthenticationConfig;
+import io.prestosql.plugin.password.file.PasswordStore;
+import io.prestosql.plugin.password.file.User;
+import io.prestosql.spi.PrestoException;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.concurrent.TimeUnit;
+
+import static io.prestosql.plugin.password.file.HashingAlgorithm.BCRYPT;
+import static io.prestosql.plugin.password.file.HashingAlgorithm.PBKDF2;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestFileAuthConfig
+{
+    private static final Logger log = Logger.get(TestFileAuthConfig.class);
+
+    private int maximumSize = 1000;
+    private Cache<PasswordStore, Boolean> bcryptCache;
+    private Cache<PasswordStore, Boolean> pbkdf2Cache;
+    private int bcryptMinCost = 8;
+    private int pbkdf2MinIterations = 1000;
+    private FileBasedAuthenticationConfig config;
+
+    @BeforeClass
```

Comment:
The Maven Surefire plugin will run TestNG methods in parallel, with a single instance of the test class. Since the test methods are modifying the `config` instance, they will stomp on each other. So, we need two fixes:

* Change to use `@BeforeMethod` and `@AfterMethod`
* Add `@Test(singleThreaded = true)` to the test class

Also, make these setup and teardown methods public, since they are called by TestNG. (it probably cheats to get around access restrictions, but logically they should be public)

Reply:
Done